gensim/models/keyedvectors.py

```diff
@@ -1695,6 +1697,79 @@ def intersect_word2vec_format(self, fname, lockf=0.0, binary=False, encoding='ut
             msg=f"merged {overlap_count} vectors into {self.vectors.shape} matrix from {fname}",
         )
 
+    def vectors_for_all(self, keys: Union[Iterable, Dictionary],
+                        allow_inference: bool = True,
+                        copy_vecattrs: bool = False) -> 'KeyedVectors':
+        """Produce vectors for all given keys as a new :class:`KeyedVectors` object.
+
+        Notes
+        -----
+        The keys will always be deduplicated. For optimal performance, you should not pass entire
+        corpora to the method. Instead, you should construct a
+        :class:`~gensim.corpora.dictionary.Dictionary` object out of your corpus first:
+
+        >>> import gensim.downloader as api
+        >>> from gensim.test.utils import common_texts
+        >>>
+        >>> model = api.load('word2vec-google-news-300')
+        >>> corpus = common_texts
+        >>> dictionary = Dictionary(corpus)  # construct a vocabulary out of your corpus
+        >>> word_vectors = model.wv.vectors_for_all(dictionary)  # create word-vectors for words in your corpus
+
+        If the keys are a :class:`~gensim.corpora.dictionary.Dictionary` object, they will be
+        sorted in a decreasing order of collection frequency (`Dictionary.cfs`) before producing
+        the new :class:`KeyedVectors` object to improve cache-warmness for subsequent operations.
+
+        Parameters
+        ----------
+        keys : {iterable, :class:`~gensim.corpora.dictionary.Dictionary`}
+            The keys that will be vectorized.
+        allow_inference : bool, optional
+            In subclasses such as :class:`~gensim.models.fasttext.FastTextKeyedVectors`,
+            vectors for out-of-vocabulary keys (words) may be inferred. Default is True.
+        copy_vecattrs : bool, optional
+            Additional attributes set via the :meth:`KeyedVectors.set_vecattr` method
+            will be preserved in the produced :class:`KeyedVectors` object. Default is False.
+            To ensure that *all* the produced vectors will have vector attributes assigned,
+            you should set `allow_inference=False`.
+
+        Returns
+        -------
+        keyedvectors : :class:`~gensim.models.keyedvectors.KeyedVectors`
+            Vectors for all the given keys.
+
+        """
+        def is_key_defined(key) -> bool:
+            if allow_inference:
+                return key in self
+            else:
+                return key in self.key_to_index
+
+        if isinstance(keys, Dictionary):
+            vocab = sorted(keys.cfs.items(), key=lambda x: (-x[1], x[0]))  # sort by decreasing collection frequency
+            vocab = (keys[key] for key, _ in vocab)
+        else:
+            vocab = keys
+
+        vocab = filter(is_key_defined, vocab)  # remove undefined keys
```

Comment:
Please avoid `filter` / `map` (anti-pattern), prefer comprehension.

In this case, the (trivial) inner function seems useless:
```suggestion
        vocab = [key in self if allow_inference else key in self.key_to_index for key in vocab]
```

Reply:
I don't personally mind the `map`/`filter` style in Python, but if something can be done in a suitably compact 1-liner, and/or not resort to custom functions defined elsewhere (even just a few lines away), I prefr the comprehension version for all-in-one-place clarity.

Reply:
Fixed in 26baf6d3.